conversant/prompt_chatbot.py

```diff
@@ -186,11 +188,103 @@ def _update_max_context_examples(
+            
+            of the previous chunk
+            
+        Returns:
+            Interaction: Dictionary of query and generated LLM response
+            
+            bool: Indicates if the generation should be stopped
+        """
+        current_prompt = self.get_current_prompt(query)
+
+        current_prompt_size = self.co.tokenize(current_prompt).length
+
+        if current_prompt_size > self.max_prompt_size:
+            max_context_examples = self._update_max_context_examples(
+                current_prompt_size, self.chatbot_config["max_context_examples"]
+            )
+            current_prompt = self.get_current_prompt(query, max_context_examples)
+
+        elif (
+            self.curr_max_context_examples
+            != self.chatbot_config["max_context_examples"]
+        ):
+            warnings.warn(
+                "The max_context_examples value returned"
+                f" to {self.chatbot_config['max_context_examples']} - "
+                f"value set in the original config"
+            )
+      
+        generated_object = self.co.generate(
+            model=self.client_config["model"],
+            prompt=current_prompt,
+            max_tokens=TOKENS_PER_REQUEST,
+            temperature=self.client_config["temperature"],
+            frequency_penalty=self.client_config["frequency_penalty"],
+            presence_penalty=self.client_config["presence_penalty"],
+            stop_sequences=self.client_config["stop_sequences"],
+        )
+        # If response was cut off by .generate() finding a stop sequence,
+        # remove that sequence from the response.
+        response = generated_object.generations[0].text
+
+        is_final_chunk = False
+        if self.should_stop(response) or response=="":
+            is_final_chunk = True
+            for stop_seq in self.client_config["stop_sequences"]:
+                if response.endswith(stop_seq):
+                    response = response[: -len(stop_seq)]
+
+        if is_from_scratch:
+            response = response.lstrip()
+            self.chat_history.append(self.prompt.create_interaction(query, response))
+            self.prompt_size_history.append(
+                self.co.tokenize(
+                    self.prompt.create_interaction_string(query, response)
+                ).length
+            )
+            self.prompt_history.append(current_prompt)
+            
+        elif response != "": 
+            self.chat_history[-1]['bot'] += response
+            self.prompt_history[-1] += response 
+            self.prompt_size_history[-1] += self.co.tokenize(response).length
+          
+        return self.chat_history[-1]['bot'], is_final_chunk   
+
     def reply(self, query: str) -> Interaction:
         """Replies to a query given a chat history.
 
         The reply is then generated directly from a call to a LLM.
 
+        Used for getting the initial response from the bot.
```

Comment:
Is this always true? I wouldn't expect the addition of our partial reply method to change the interface of the reply() method or its expected use cases -- e.g., for very fast models or for small values of max_tokens, it may be fine to still use reply() always.

Reply:
Sounds good, will remove this